vktrace/vktrace_common/vktrace_trace_packet_utils.c

```diff
@@ -113,22 +114,53 @@ uint64_t vktrace_get_time() {
 uint64_t vktrace_get_time() { return 0; }
 #endif
 
-//=============================================================================
-// trace file header
-
-vktrace_trace_file_header* vktrace_create_trace_file_header() {
-    vktrace_trace_file_header* pHeader = VKTRACE_NEW(vktrace_trace_file_header);
-    memset(pHeader, 0, sizeof(vktrace_trace_file_header));
-    pHeader->trace_file_version = VKTRACE_TRACE_FILE_VERSION;
-    vktrace_gen_uuid(pHeader->uuid);
-    pHeader->trace_start_time = vktrace_get_time();
+uint64_t get_endianess() {
+    uint32_t x = 1;
+    return *((char*)&x) ? VKTRACE_BIG_ENDIAN : VKTRACE_LITTLE_ENDIAN;
+}
 
-    return pHeader;
+uint64_t get_arch() {
+    uint64_t rval = 0;
+#if defined(PLATFORM_LINUX)
```

Comment:
Yes, the Android build defines PLATFORM_LINUX and tries to inherit most sections unchanged.